src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -295,7 +300,27 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
     	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+		JenkinsAuthenticationProvider checkAppScan360Connection = new JenkinsAuthenticationProvider(m_credentials,build.getParent().getParent());
+        if(m_type.equals("Dynamic Analyzer") && checkAppScan360Connection.isAppScan360()){
+			throw new AbortException(Messages.error_dynamic_analyzer_AppScan360());
+		}
+
+        if(!checkAppScan360Connection.isAppScan360()){
+            if(m_authProvider.getacceptInvalidCerts()){
+                progress.setStatus(new Message(1,Messages.warning_asoc_certificates()));
+            }
+        }
+
+		if(m_intervention && checkAppScan360Connection.isAppScan360()){
+			progress.setStatus(new Message(1,Messages.warning_allow_intervention_AppScan360()));
+		}
+
+		if(getScanProperties(build, listener).get("openSourceOnly") !=null){
```

Comment:
If you look at the definition of the `getScanProperties()` method, it does a bit of work to construct the properties, including resolving variables, etc. So we should avoid having to call it multiple times unless it's necessary. With these changes, we're now calling that method 3 times within this single function. To avoid that, I suggest calling it once at the beginning of the function and storing the value in a local variable.

Reply:
Okay, sure.